sky/provision/kubernetes/network_utils.py

```diff
@@ -0,0 +1,91 @@
+from typing import Dict, Tuple
+
+from sky.adaptors import kubernetes
+
+
+def create_namespaced_ingress(namespace: str, ingress_name: str,
```

Comment:
Jupyter does not seems to play nicely with ingress:

```yaml
#jupyter.yaml
resources:
  ports: 8888

setup: pip install jupyter

run: jupyter lab --port 8888 --no-browser --ip=0.0.0.0
```
On accessing `https://35.222.205.70/skypilot/test-2ea4_8888/` in Chrome, I get 404:

<img width="983" alt="image" src="https://github.com/skypilot-org/skypilot/assets/4416605/95260f24-f3ab-41b2-9112-bac5674e7683">

Note that it tried to redirect me to https://35.222.205.70/lab, which is incorrect.

On accessing the exact suburl (`/lab?token=....`), I get an empty white page but loading some js/css is likely failing with 404:
<img width="2028" alt="image" src="https://github.com/skypilot-org/skypilot/assets/4416605/f379ebad-6668-4f59-8229-779b5de9d5be">

Can we configure the ingress to correctly redirect to the sub-path we have under the ingress?